include/zephyr/logging/log.h

```diff
@@ -72,6 +72,24 @@ extern "C" {
  */
 #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
 
+/**
+ * @brief Writes a WARNING level message to the log on the first execution only.
+ *
+ * @details It's meant for situations that warrant investigation but could clutter
+ * the logs if output on every execution.
+ *
+ * @param ... A string optionally containing printk valid conversion specifier,
+ * followed by as many values as specifiers.
+ */
+#define LOG_WRN_ONCE(...)					\
+	do {							\
+		static int __warned;				\
```

Comment:
What about using some smaller data type?

Reply:
I was trying to avoid pulling in other headers but I see `<stdint.h>` types are already being used.